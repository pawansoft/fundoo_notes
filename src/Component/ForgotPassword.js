import React, { Component } from 'react';
import {
    TextInput,
    View,
    TouchableOpacity,
    Text,
    Image,
    ScrollView
} from 'react-native';

import UserService, { ResetPasscodeService } from '../../Services/UserServices/UserService';
import login_style from '../Style/login_style';
import reset_component_style from '../Style/reset_component_style';
import DialogeBox from './DialogeBox';
export default class ForgotPassword extends Component {

    constructor(props) {
        super(props);
        this.state = {
            isEmailValid: '',
            emailId: '',
        }
    }

    findValidEmail = async () => {
        const regex = /^[0-9a-zA-Z]+([._+-][0-9a-zA-Z]+)*[@][0-9A-Za-z]+([.][a-zA-Z]{2,4})*$/
        if (regex.test(this.state.emailId) == false) {
            await this.setState({
                isEmailValid: 'Invalid Email'
            })
        } else {
            await this.setState({
                isEmailValid: ''
            })
        }
    }

    handleEmailId = async (emailId) => {
        await this.setState({
            emailId: emailId
        })
    }

    handleResetButton = () => {
        if (
            this.state.emailId != '' &&
            this.state.isEmailValid == '') {
                UserService.ResetPasscodeService(this.state.emailId).then(() => {
                    alert('Password reset mail has sent to you please check your mail')
                })
                .catch(error => console.log(error))     
        }
        else {
            alert('Oops something went wrong')
        }
    }

    handleLoginButton = () => {
        this.props.navigation.navigate('Login')
    }

    render() {
        return (
            <View>
                <View style={login_style.header_image}>
                    <Image source={require('../assets/logo.png')} />
                </View>

                <ScrollView style={login_style.scroll_view}>
                    <View style={login_style.container}>
                        <Text style={{ alignSelf: 'center', fontWeight: 'bold' }}>Reset Password</Text>
                        
                    <View style={reset_component_style.text_container}>
                        <TextInput
                            value={this.state.emailId}
                            onChangeText={this.handleEmailId}
                            placeholder={'Email Id'}
                            onEndEditing={this.findValidEmail} />
                        <Text style={login_style.error_text}>{this.state.isEmailValid}</Text>
                    </View>

                    <View style={{ flexDirection: 'row' }}>
                        <TouchableOpacity
                            style={login_style.signup_button_container}
                            onPress={this.handleResetButton}>
                            <Text style={login_style.button_text}>
                                Next
                            </Text>
                        </TouchableOpacity>


                        <TouchableOpacity
                            style={{ marginLeft: '30%', marginTop: '7%' }}
                            onPress={this.handleLoginButton}>
                            <Text style={login_style.button_text}>
                                Go to Login
                            </Text>
                        </TouchableOpacity>
                    </View>
                </View>
                </ScrollView>
            </View>
        )
    }
}
