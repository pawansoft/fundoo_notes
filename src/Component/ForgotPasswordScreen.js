import React from 'react';
import ForgotPassword from './ForgotPassword';

const ForgotPasswordScreen = ({navigation}) => {
    
    return(
        <ForgotPassword navigation = {navigation}/>
    )
}
export default ForgotPasswordScreen;