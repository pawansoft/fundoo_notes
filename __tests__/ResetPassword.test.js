import React from 'react';
import Adapter from 'enzyme-adapter-react-16'
import {shallow, configure} from 'enzyme'
import ForgotPasswordScreen from '../src/Component/ForgotPasswordScreen';
import ForgotPassword from '../src/Component/ForgotPassword';
configure({adapter: new Adapter()})

describe('Reset Password Test', () => {
    it('ProvideRender_WhenMatchToSnapshot_TestShouldPass', () => {
        const component = shallow(<ForgotPassword/>)
        expect(component).toMatchSnapshot()
    })
})
describe('Test State of Reset Password', () => {
    test('ProvideUserName_WhenPassedWithField_ShouldStoreAtState', () =>{
        const component = shallow(<ForgotPassword/>);
        component.instance().handleUserName('pk.soft29@gmail.com');
        expect(component.instance().state.userName).toBe('pk.soft29@gmail.com')
    }),
    test('ProvideUserName_WhenPassedWithField_ShouldStoreAtState', () =>{
        const component = shallow(<ForgotPassword/>);
        component.instance().handlePassword('Pk@12345678');
        expect(component.instance().state.password).toBe('Pk@12345678')
    })
})

describe('Test Invalid Detail', () => {
    test('ProvideUserName_WhenIncorrectFormate_ShouldGenerateError', () =>{
        const component = shallow(<ForgotPassword/>);
        component.instance().handleUserName('pk');
        component.instance().validateUserName();
        expect(component.instance().state.userNameValid).toBe('Invalid Email')
    }),
    test('ProvideUserName_WhencorrectFormate_ShouldRemoveAnError', () =>{
        const component = shallow(<ForgotPassword/>);
        component.instance().handleUserName('pk');
        component.instance().validateUserName();
        expect(component.instance().state.userNameValid).toBe('Invalid Email')
        component.instance().handleUserName ('pk.soft29@gmail.com');
        component.instance().validateUserName();
        expect(component.instance().state.userNameValid).toBe('');
    }),
    test('ProvideUserName_WhenPasswordIsInvalid_ShouldGenerateError', () =>{
        const component = shallow(<ForgotPassword/>);
        component.instance().handlePassword('12345678');
        component.instance().validatePassword();
        expect(component.instance().state.passwordValid).toBe('Not Strong')
    })

    test('ProvideUserName_WhenPasswordIsValid_ShouldStoreAtState', () =>{
        const component = shallow(<ForgotPassword/>);
        component.instance().handlePassword('2345678');
        component.instance().validatePassword();
        expect(component.instance().state.passwordValid).toBe('Not Strong');
        component.instance().handlePassword('Pk@16123114');
        component.instance().validatePassword();
        expect(component.instance().state.passwordValid).toBe('')
    })
})