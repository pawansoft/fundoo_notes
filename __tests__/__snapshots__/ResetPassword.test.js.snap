// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reset Password Test ProvideRender_WhenMatchToSnapshot_TestShouldPass 1`] = `ShallowWrapper {}`;
